doc/kubernetes/modules/ROOT/pages/running/deployments/active-passive-sync.adoc

```diff
@@ -0,0 +1,214 @@
+Some requests might receive an error message or are delayed for some seconds.
+Manual operations might be necessary depending on the database.
+| No data loss ^3^
+| Seconds to minutes (depending on the database)
+
+| Primary Datacenter
+| If none of the Keycloak nodes is available, the loadbalancer will detect the outage and redirect the traffic to the secondary site.
+Some requests might receive an error message while the loadbalancer hasn't detected the primary datacenter failure.
+The setup will be degraded until the primary site is back up and the session state has been manually synced from the secondary to the primary site.
+| No data loss^3^
+| Less than one minute
+
+| Secondary Datacenter
+| If the secondary datacenter is not available, it will take a moment for the primary Infinispan and database to mark the secondary datacenter offline.
+Some requests might receive an error message while the detection takes place.
+Once the secondary datacenter is up again, the session state needs to be manually synced from the primary site to the secondary site.
+| No data loss^3^
+| Less than one minute
+
+|===
+
+^1^: Recovery point objective, assuming all parts of the setup were healthy at the time this occurred. +
+^2^: Recovery time objective. +
+^3^: Manual operations needed to restore the degraded setup.
+
+The statement "`No data loss`" depends on the setup not being degraded from previous failures, which includes completing any pending manual operations to resynchronize the state between the datacenters.
+
+=== Known limitations
+
+==== Upgrades
+
+* On Keycloak major version upgrade, all session data (except offline session) might be lost as the data format is not guaranteed to be compatible between major versions.
+* On any Infinispan upgrade (major, minor or patch), all session data might be lost as cross-DC communication is guaranteed to work only between identical versions of Infinispan.
+
+==== Failovers
+
+* A successful failover requires a setup not degraded from previous failures.
+All manual operations like a re-synchronization after a previous failure must be complete to prevent a data loss.
+Customers must use monitoring to ensure degradations are detected and handled in a timely manner.
+
+==== Switchovers
+
+* A successful switchover requires a setup not degraded from previous failures.
+All manual operations like a re-synchronization after a previous failure must be complete to prevent a data loss.
+Customers must use monitoring to ensure degradations are detected and handled in a timely manner.
+
+==== Out-of-sync datacenters
+
+* The datacenters can become out of sync when a synchronous Infinispan request fails.
+This is currently difficult to monitor, and it would need a full manual re-sync of Infinispan to recover.
+Monitoring the number of cache entries in both datacenters and Keycloak's log file can show when this would become necessary.
+Future versions of Keycloak and Infinispan plan to automate this.
```

Comment:
What about out-of-sync databases? Is that handled automatically?

Reply:
Let's assume for now this is handled by the database. At least for Aurora it would ensure that there is only one writer database at a time. Other databases might need extra attention there, but we would need to figure that out only when we add more databases to the blueprint.